pkg/auth/certs/provider.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package certs
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+)
+
+type CertificateProvider interface {
+	// GetTrustBundle gives the CA trust bundle for the certificate provider
+	// this is then used to verify the certificates given by the peer in the handshake
+	GetTrustBundle() (*x509.CertPool, error)
+
+	// GetCertificateForIdentity gives the certificate and intermediates required
+	// to send as trust chain for a certain identity as well as a private key
+	GetCertificateForIdentity(identity string) (*tls.Certificate, error)
+
+	// ValidateSAN will check if the SANs are valid for the given identity
+	// this function is needed as SPIFFE encodes the full ID in the URI SAN
+	ValidateSAN(identity string, cert *x509.Certificate) (bool, error)
```

Comment:
Should this be called `ValidateIdentity`, just in case we have something that doesn't use the SAN?

Reply:
Good point, i did wanted to kind of make the point here that it did not do the CA chain validation, identity may be a good term